linux_os/guide/system/permissions/restrictions/enable_execshield_settings/sysctl_kernel_exec_shield/bash/shared.sh

```diff
@@ -19,7 +19,7 @@ if [ "$(getconf LONG_BIT)" = "64" ] ; then
   if grep --silent noexec /boot/grub2/grub*.cfg ; then 
         sed -i "s/noexec.*//g" /etc/default/grub
         sed -i "s/noexec.*//g" /etc/grub.d/*
-        GRUBCFG=/boot/grub2/*.cfg
+        GRUBCFG="/boot/grub2/*.cfg"
```

Comment:
This doesn't solve the problem - the assumption is that there is exactly one file ending with `.cfg`, and it truly won't work if there are more files like that. If you quote it, the asterisk won't be expanded, so it will break completely.
From the top of the `if` statement, it looks like that we may be after `/boot/grub2/grub.cfg` or `/boot/grub2/grub2.cfg`, so it may make sense to determine the config file name early in the remediation, and using it as a simple variable later on.
It can be a for loop that iterates through all occurrences of `/boot/grub2/*.cfg`, assigns the first one to that variable, and throws an error if there are more occurrences.

Reply:
I agree with your statements. Also I wanted to point out that the OVAL check only checks for a single file:

https://github.com/ComplianceAsCode/content/blob/master/linux_os/guide/system/permissions/restrictions/enable_execshield_settings/sysctl_kernel_exec_shield/oval/shared.xml#L60

Reply:
I'll remove this piece of code since it needs a little bit more of developing so we can merge the other fixes right away.

Reply:
I am fine with the remediation being in line with the OVAL check.